syntax = "proto3";
package vectorbase;

// configuration for the HNSW index which will be build next
message IndexConfig {
  int32 M = 1;
  int32 ef_construction = 2;
  string metric = 3;
}

// record on storing a vector
message VectorRecord {
  int64 id = 1;
  repeated float vector = 2;
  map<string, string> metadata = 3;
}

// segment config for flush/partitioning
message SegmentConfig {
  int32 max_vectors_per_segment = 1;
  int32 dim = 2;
}
